tools/run_tests/xds_k8s_test_driver/tests/outlier_detection_test.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2022 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from typing import List
+
+from absl import flags
+from absl.testing import absltest
+
+from framework import xds_k8s_testcase
+from framework import xds_url_map_testcase
+from framework.infrastructure import k8s
+from framework.test_app import server_app
+
+logger = logging.getLogger(__name__)
+flags.adopt_module_key_flags(xds_k8s_testcase)
+
+# Type aliases
+RpcTypeUnaryCall = xds_url_map_testcase.RpcTypeUnaryCall
+RpcTypeEmptyCall = xds_url_map_testcase.RpcTypeEmptyCall
+_XdsTestServer = xds_k8s_testcase.XdsTestServer
+_XdsTestClient = xds_k8s_testcase.XdsTestClient
+
+
+class OutlierDetectionTest(xds_k8s_testcase.RegularXdsKubernetesTestCase):
+
+    def test_outlier_detection(self) -> None:
+        REPLICA_COUNT = 5
+
+        with self.subTest('00_create_health_check'):
+            self.td.create_health_check()
+
+        with self.subTest('01_create_backend_service'):
+            self.td.create_backend_service(
+                outlier_detection={
+                    'interval': {
+                        'seconds': 2,
+                        'nanos': 0
+                    },
+                    'successRateRequestVolume': 20
+                })
+
+        with self.subTest('02_create_url_map'):
+            self.td.create_url_map(self.server_xds_host, self.server_xds_port)
+
+        with self.subTest('03_create_target_proxy'):
+            self.td.create_target_proxy()
+
+        with self.subTest('04_create_forwarding_rule'):
+            self.td.create_forwarding_rule(self.server_xds_port)
+
+        test_servers: List[_XdsTestServer]
+        with self.subTest('05_start_test_servers'):
+            test_servers = self.startTestServers(replica_count=REPLICA_COUNT)
+
+        with self.subTest('06_add_server_backends_to_backend_services'):
+            self.setupServerBackends()
+
+        test_client: _XdsTestClient
+        with self.subTest('07_start_test_client'):
+            test_client = self.startTestClient(test_servers[0], qps=100)
+
+        with self.subTest('08_test_client_xds_config_exists'):
+            self.assertXdsConfigExists(test_client)
+
+        with self.subTest('09_test_servers_received_rpcs_from_test_client'):
+            self.assertRpcsEventuallyGoToGivenServers(test_client, test_servers)
+
+        rpc_types = [RpcTypeUnaryCall]
```

Comment:
nit: if the list is not going to be modified, prefer a tuple
```suggestion
        rpc_types = (RpcTypeUnaryCall, )
```
```suggestion
        rpc_types = [RpcTypeUnaryCall]
```